src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -6872,4 +6872,7 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="WRN_CompileTimeCheckedOverflow_Title" xml:space="preserve">
     <value>The operation may overflow at runtime (use 'unchecked' syntax to override)</value>
   </data>
+  <data name="ERR_NoAwaitOnAsyncEnumerable" xml:space="preserve">
+    <value>If you intend to iterate over an IAsyncEnumerable, then put the await statement before the loop statement. i.e. await foreach.</value>
```

Comment:
>If you intend to iterate over an IAsyncEnumerable, then put the await statement before the loop statement. i.e. await foreach. [](http://example.com/codeflow?start=11&length=126)

How about keeping the error portion of the message but change the suggestion portion?

`'{0}' does not contain a definition for '{1}' and no accessible extension method '{1}' accepting a first argument of type '{0}' could be found (did you mean to iterate over the async collection with `await foreach`?)`